internal/provider/vaultsecrets/resource_vault_secrets_integration_postgres.go

```diff
@@ -0,0 +1,284 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vaultsecrets
+
+import (
+	"context"
+	"fmt"
+	"maps"
+
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/stable/2023-11-28/client/secret_service"
+	secretmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/stable/2023-11-28/models"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+	"github.com/hashicorp/terraform-provider-hcp/internal/provider/modifiers"
+)
+
+type IntegrationPostgres struct {
+	// Input fields
+	ProjectID               types.String `tfsdk:"project_id"`
+	Name                    types.String `tfsdk:"name"`
+	Capabilities            types.Set    `tfsdk:"capabilities"`
+	StaticCredentialDetails types.Object `tfsdk:"static_credential_details"`
+
+	// Computed fields
+	OrganizationID types.String `tfsdk:"organization_id"`
+	ResourceID     types.String `tfsdk:"resource_id"`
+	ResourceName   types.String `tfsdk:"resource_name"`
+
+	// Inner API-compatible models derived from the Terraform fields
+	capabilities            []*secretmodels.Secrets20231128Capability                     `tfsdk:"-"`
+	staticCredentialDetails *secretmodels.Secrets20231128PostgresStaticCredentialsRequest `tfsdk:"-"`
+}
+
+type connectionString struct {
+	ConnectionString types.String `tfsdk:"connection_string"`
+}
+
+var _ resource.Resource = &resourceVaultSecretsIntegrationPostgres{}
+var _ resource.ResourceWithConfigure = &resourceVaultSecretsIntegrationPostgres{}
+var _ resource.ResourceWithModifyPlan = &resourceVaultSecretsIntegrationPostgres{}
+var _ resource.ResourceWithImportState = &resourceVaultSecretsIntegrationPostgres{}
+
+func NewVaultSecretsIntegrationPostgresResource() resource.Resource {
+	return &resourceVaultSecretsIntegrationPostgres{}
+}
+
+type resourceVaultSecretsIntegrationPostgres struct {
+	client *clients.Client
+}
+
+func (r *resourceVaultSecretsIntegrationPostgres) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_vault_secrets_integration_postgres"
+}
```

Comment:
@dhuckins there is an initiative on the Ecosystem side to make the integration management API more generic and reduce the amount of provider-specific code necessary to build and maintain for new integrations. We,ll move closer to how the rotating & dynamic secret resources are architected.

We will shortly implement a `hcp_vault_secrets_integration` resource ([ticket](https://hashicorp.atlassian.net/browse/HV-3178)) to support all providers and deprecate the provider-specific resources. We should not introduce new provider-specific integration resources e.i. `hcp_vault_secrets_integration_{provider}`.

We should wait a bit and support postgres only via the `hcp_vault_secrets_integration` resource. Sorry if this is last minute and requires work on this PR. If there are any questions please reach out to the team on #team-hvs-ecosystem.

Reply:
will close this then. 
have a gateway PR that I can put up